### PR TITLE
Parachains shared.rs to Frame V2

### PR DESCRIPTION
**Pull request by ferrell-code**
relates: #2882 

Following the upgrade guidelines here: https://crates.parity.io/frame_support/attr.pallet.html#upgrade-guidelines.

# :warning: Breaking Change :warning: 
From https://crates.parity.io/frame_support/attr.pallet.html#checking-upgrade-guidelines

 >storages now use PalletInfo for module_prefix instead of the one given to decl_storage: Thus any use of this pallet in construct_runtime! should be careful to update name in order not to break storage or to upgrade storage (moreover for instantiable pallet). If pallet is published, make sure to warn about this breaking change.

So users of the `ParasShared` pallet must be careful about the name they used in `construct_runtime!.` Hence the runtime-migration label, which might not be needed depending on the configuration of the `ParasShared` pallet.

Kusama and westend both already use `ParasShared` in `construct_runtime!` no migration needed. Polkadot does not use the `ParasShared` pallet.

Rococo uses `Shared` in `construct_runtime!`, therefore I altered it to `ParasShared` to keep the pallet name the same as seen in `decl_storage!` of `ParasShared`

### CODE REVIEW

File: runtime/test-runtime/src/lib.rs [hunk 1]
```diff
@@ -532,6 +532,7 @@ construct_runtime! {
 		ParasInherent: parachains_paras_inherent::{Pallet, Call, Storage, Inherent},
 		Initializer: parachains_initializer::{Pallet, Call, Storage},
 		Paras: parachains_paras::{Pallet, Call, Storage, Origin, Event},
+		ParasShared: parachains_shared::{Pallet, Call, Storage},
 		Scheduler: parachains_scheduler::{Pallet, Call, Storage},
 		ParasSudoWrapper: paras_sudo_wrapper::{Pallet, Call},
 		SessionInfo: parachains_session_info::{Pallet, Call, Storage},
```
Comment (ferrell-code) [inline runtime/test-runtime/src/lib.rs:538]: pallet must be put in `construct_runtime` for tests... if you make a call into `ParasShared` storage without doing so frame_support will panic as it needs a pallet name (the one that was previously declared in `decl_storage!`. Probably should check if any of cumulus tests do this

**Comment by thiolliere** [comment]
looking at the number of line removed, I guess some master merge got wrong, isn't it ? @ferrell-code

**Comment by ferrell-code** [comment]
Mr. Tabrizi swooping in with those 5 am pushes lol, I force pushed back to the original point haha and resoolved conflicts

**Review by KiChjang — APPROVED** [review]

**Review by emostov — APPROVED** [review]

**Comment by KiChjang** [comment]
bot merge

**Comment by shawntabrizi** [comment]
@ferrell-code yeah sorry about that.

Was gonna fix it for you, but then got too tired, and gave up lol

**Comment by ferrell-code** [comment]
I got a kick out of it, always appreciate your help :dromedary_camel:

**Pull request merged**